apps/vs-code-designer/src/app/commands/deploy/hybridLogicApp/index.ts

```diff
@@ -88,6 +101,157 @@ export const deployHybridLogicApp = async (context: IActionContext, node: SlotTr
+      return; // FQDN is available
+    }
+
+    try {
+      // Refresh the hybrid site details
+      node.hybridSite = await clientContainer.containerApps.get(node.resourceGroupName, node.hybridSite.name);
+    } catch (error) {
+      console.error(`Error refreshing hybrid site details: ${error.message}`);
+    }
+
+    // Wait before retrying
+    await new Promise((resolve) => setTimeout(resolve, delay));
+  }
+
+  throw new Error('Failed to retrieve ingress FQDN after waiting.');
+};
+
+const createZipFileOnDisk = async (sourceDir: string): Promise<string> => {
+  return new Promise((resolve, reject) => {
+    const tempDir = os.tmpdir();
+    const zipFilePath = path.join(tempDir, `logicapp-deploy-${Date.now()}.zip`);
+    const zipFile = new yazl.ZipFile();
+
+    // Add all files in the directory to the zip
+    const addDirectoryToZip = (dir: string, basePath: string) => {
+      const files = fs.readdirSync(dir);
+      for (const file of files) {
+        const fullPath = path.join(dir, file);
+        const relativePath = path.relative(basePath, fullPath);
+
+        if (fs.statSync(fullPath).isDirectory()) {
+          addDirectoryToZip(fullPath, basePath); // Recursively add subdirectories
+        } else {
+          zipFile.addFile(fullPath, relativePath); // Add file to the zip
+        }
+      }
+    };
+
+    addDirectoryToZip(sourceDir, sourceDir);
+
+    // Write the zip file to disk
+    const output = fs.createWriteStream(zipFilePath);
+    zipFile.outputStream.pipe(output);
+
+    output.on('close', () => resolve(zipFilePath));
+    output.on('error', (err) => reject(err));
+    zipFile.end();
+  });
+};
+
+const getAccessTokenForZipDeploy = async (node: SlotTreeItem, context: ILogicAppWizardContext): Promise<string> => {
+  // Retrieve environment variables for AAD
+  const tenantId = node.hybridSite.template.containers[0]?.env.find((e) => e.name === workflowAppAADTenantId)?.value;
+  const clientId = node.hybridSite.template.containers[0]?.env.find((e) => e.name === workflowAppAADClientId)?.value;
+  const clientSecretKey = node.hybridSite.template.containers[0]?.env.find((e) => e.name === workflowAppAADClientSecret)?.secretRef;
+  const clientSecret = node.resourceTree?.hybridSiteSecrets?.find((s) => s.name === clientSecretKey)?.value ?? context.aad?.clientSecret;
+
+  if (!tenantId || !clientId || !clientSecret) {
+    throw new Error(
+      `Missing required environment variables: ${workflowAppAADClientId}, ${workflowAppAADTenantId}, ${workflowAppAADClientSecret}`
+    );
+  }
+
+  // Create a credential using ClientSecretCredential
+  const credential = new ClientSecretCredential(tenantId, clientId, clientSecret);
+
+  // Get the access token for the application
+  const tokenResponse = await credential.getToken(`api://${clientId}/.default`);
+
+  if (!tokenResponse || !tokenResponse.token) {
+    throw new Error('Failed to retrieve access token for zip deploy API');
```

Comment:
Can we localize the error and also log the erro in the 
```typescript
context.telemetry.properties.getAccessTokenError = errorMessage;
```
getAccessTokenError its just an example it can be any value